rust/abacus-core/src/chain.rs

```diff
@@ -49,53 +54,199 @@ impl From<&'_ Address> for ethers::types::H160 {
     }
 }
 
-/// Quick single-use macro to prevent typing domain and chain twice and risking
-/// inconsistencies.
-macro_rules! domain_and_chain {
-    {$($domain:literal <=> $chain:literal,)*} => {
-        /// Get the chain name from a domain id. Returns `None` if the `domain` is unknown.
-        pub fn chain_from_domain(domain: u32) -> Option<&'static str> {
-            match domain {
-                $( $domain => Some($chain), )*
-                _ => None
-            }
+/// All mainnet domains supported by Abacus.
```

Comment:
I added these to add some compile-time checks on ensuring we always have a coingecko API ID for each mainnet chain. I'm curious how you feel about this vs the old macro

If this feels really weird to you I'm happy to get rid of this. Another idea is instead of having separate enums for mainnet / testnet / local test chains, is to just have one big enum with them all, and then have a runtime check for getting coingecko API IDs